webplugin/js/app/km-widget-custom-events.js

```diff
@@ -0,0 +1,100 @@
+/**
+ // Kommunicate custome events available for user for the specific event in chat widget.
+ // resp - JSON object containing eventAction and other property exposed to user.
+ // defination of function can be overwrite by user through subscribeToEvents function.
+ */
+kmWidgetEvents = {
+    onChatWidgetOpen: function (resp) {},
+    onChatWidgetClose: function (resp) {},
+    onFaqClick: function (resp) {},
```

Comment:
call the GA code from here

onFaqClick: function() { 
    sendEventToGoogleAnalytics(eventMapping['FAQ Menu']);
}

Reply:
These function definitions are getting overwritten by the user's custom event function definition, In case we call `sendEventToGoogleAnalytics(eventMapping['FAQ Menu'])` from here will be overwritten by the user code. And what if user also wants to send events to GoogleAnalytics as well as any other website?

Reply:
got it
found the right file, goto mck-sidebox-1.0.js

   _this.subscribeToEvents is a common function where all events are coming, i think here we should write code to call the GA related functions.
This is a common place for all events

Reply:
So i think GA has zero dependencies on the chat widget custom events and `_this.subscribeToEvents` is called whenever the user call `Kommunicate.subscribeToEvents(events)` in on init function. I think this is not the right place to put GA-related code here?
What's your opinion?

Reply:
ya right